src/runtime_src/core/tools/xbmgmt2/SubCmdConfigure.cpp

```diff
@@ -0,0 +1,503 @@
+      continue;
+    }
+    if (!key.first.compare("scaling_temp_override")) {
+      xrt_core::device_update<xrt_core::query::xmc_scaling_temp_override>(_dev.get(), key.second.get_value<std::string>());
+      continue;
+    }
+    throw std::runtime_error(boost::str(boost::format("'%s' is not a supported config entry") % key.first));
+  }
+}
+
+static config
+get_daemon_conf()
+{
+  config cfg;
+  cfg.host = xrt_core::get_hostname();
+
+  std::ifstream istr(config_file);
+  if (!istr)
+    return cfg;
+
+  // Load persistent value, may overwrite default one
+  for (std::string line; std::getline(istr, line);) {
+    auto pos = line.find('=', 0);
+    if (pos == std::string::npos)
+      throw xrt_core::system_error(EIO, "Bad daemon config file line '" + line + "'");
+    auto key = line.substr(0, pos++);
+    auto value = line.substr(pos);
+
+    if (key == "host") 
+      cfg.host = value;
+  }
+
+  return cfg;
+}
+
+/*
+ * helper function for option: showx
+ * shows daemon config
+ */
+static void 
+show_daemon_conf()
+{
+  auto cfg = get_daemon_conf();
+  std::cout << "Daemon:" << std::endl;
+  std::cout << "\t";
+  cfg.write(std::cout);
+}
+
+/*
+ * helper function for option: showx
+ * shows device config
+ */
+static void 
+show_device_conf(xrt_core::device* device)
+{
+  auto bdf_raw = xrt_core::device_query<xrt_core::query::pcie_bdf>(device);
+  auto bdf = xrt_core::query::pcie_bdf::to_string(bdf_raw);
+
+  std::cout << bdf << "\n";
+
+  bool is_mfg = false;
+  bool is_recovery = false;
+  try {
+    is_mfg = xrt_core::device_query<xrt_core::query::is_mfg>(device);
+    is_recovery = xrt_core::device_query<xrt_core::query::is_recovery>(device);
+  }
+  catch (const std::exception& ex) {
+    std::cerr << ex.what() << "\n";
+  }
+
+  if (is_mfg || is_recovery)
+    throw xrt_core::error(std::errc::operation_canceled, "This operation is not supported with manufacturing image.\n");
+
+  try {
+    auto sec_level = xrt_core::device_query<xrt_core::query::sec_level>(device);
+    std::cout << boost::format("  %-33s: %s\n") % "Security level" % sec_level;
+  }
+  catch (xrt_core::query::exception&) {
+    //safe to ignore. These sysfs nodes are not present for vck5000 
+  }
+
+  try {
+    auto scaling_enabled = xrt_core::device_query<xrt_core::query::xmc_scaling_enabled>(device);
+    std::cout << boost::format("  %-33s: %s\n") % "Runtime clock scaling enabled" % scaling_enabled;
+  }
+  catch (xrt_core::query::exception&) {
+    //safe to ignore. These sysfs nodes are not present for u30
+  }
+
+  try {
+    auto scaling_override = xrt_core::device_query<xrt_core::query::xmc_scaling_power_override>(device);
+    std::cout << boost::format("  %-33s: %s\n") % "Scaling threshold power override" % scaling_override;
+  }
+  catch (xrt_core::query::exception&) {
+    //safe to ignore. These sysfs nodes are not present for u30
```

Comment:
O.K. - But what happens if we are NOT on an U30.

Reply:
Same as above, the try logic executes